src/Pos/Block/Logic/Internal.hs

```diff
@@ -88,23 +95,23 @@ applyBlocksUnsafeDo
 applyBlocksUnsafeDo blunds pModifier = do
     -- Note: it's important to put blocks first
     mapM_ putToDB blunds
+    TxpGlobalSettings {..} <- ncTxpGlobalSetttings <$> getNodeContext
     usBatch <- SomeBatchOp <$> usApplyBlocks blocks pModifier
     delegateBatch <- SomeBatchOp <$> delegationApplyBlocks blocks
-#ifdef WITH_EXPLORER
-    curTime <- currentTimeSlotting
-    txBatch <- txApplyBlocks blunds curTime
-#else
-    txBatch <- txApplyBlocks blunds
-#endif
+    txpBatch <- tgsApplyBlocks $ map toTxpBlund blunds
     sscApplyBlocks blocks Nothing -- TODO: pass not only 'Nothing'
     let putTip = SomeBatchOp $
                  GS.PutTip $
                  headerHash $
                  NE.last $
                  getOldestFirst blunds
-    GS.writeBatchGState [putTip, delegateBatch, usBatch, txBatch, forwardLinksBatch, inMainBatch]
+    GS.writeBatchGState [putTip, delegateBatch, usBatch, txpBatch, forwardLinksBatch, inMainBatch]
     sscNormalize
+#ifdef WITH_EXPLORER
```

Comment:
1. It is impossible, because `eTxNormalize` works in `MonadTxpMem ExplorerExtra`, while `txNormalize` works in `MonadTxpMem ()`.
2. Note that local data processing is not abstracted yet, eventually this normalization will be part of `TxpLocalSettings`.